File: abc126_c.cpp
// https://atcoder.jp/contests/abc125/tasks/abc126_c
#include<iostream>
#include<algorithm>
//#include<string>
#include<vector>
//#include<map>
//#include<tuple>
//#include<queue>
//#include<regex>

// #include <bits/stdc++.h>
using namespace std;

int main() {
	long long N, K;
	cin >> N >> K;
	double ans = 0;

	if (N > K) ans += (double)(N - K);
	for (long long i = 1; i <= min(N, K); i++)
	{
		int logk = ceil(log2((double)K / i));
		ans += (double)pow(0.5, logk);
	}

	printf("%.12lf\n", ans / N);
	// cout << ans << endl;
	return 0;
}

File: atcoder/abc126_c.py
# C - Dice and Coin
# https://atcoder.jp/contests/abc126/tasks/abc126_c


import math

N, K = map(int, input().split())
ans = 0

if N > K:
    ans += (N - K) / N

for i in range(min(N, K)):
    logk = math.ceil(math.log2(K / (i+1)))
    ans += 1 / N * (0.5 ** logk)

print(ans)
